### PR TITLE
Make login / signing page load fast and fix redirect to app (#1516 #1567)

### PR DESCRIPTION
**Pull request by marcinbodnar**

### CODE REVIEW

**Comment by marcinbodnar** [comment]
@vgrichina I refactored previously implemented main menu hiding, so now it will depend on Redux state (instead of checking URL). 

Sign page will load faster, and the Login page will load a lot faster - for Login flow, I removed balance checking (about 1s faster) and also added functionality to fetch limited data for Account (about 0.3s faster). 

Added new reducer `flowLimitation` that will point what data should be fetched and what part of the menu and submenu are available while on Login and Sign flow.

File: src/actions/account.js [hunk 1]
```diff
@@ -6,14 +6,20 @@ import { TwoFactor } from '../utils/twoFactor'
...
+    WALLET_RECOVER_ACCOUNT_URL, 
+    setKeyMeta,
+    MULTISIG_MIN_PROMPT_AMOUNT
 } from '../utils/wallet'
 import { PublicKey, KeyType } from 'near-api-js/lib/utils/key_pair'
 import { WalletError } from '../utils/walletError'
 import { utils } from 'near-api-js'
 import { BN } from 'bn.js'
 import { showAlert, dispatchWithAlert } from '../utils/alerts'
+import { handleflowLimitation, handleClearflowLimitation } from './flowLimitation'
```
Comment (vgrichina) [inline src/actions/account.js:22]: nit: seems like Flow should be capitalized in these

**Comment by vgrichina** [comment]
@marcinbodnar looks good and already is much faster.

However I think it is necessary to do a follow-up PR to avoid loading staked balances, as this can be slow query depending on the situation.  Generally we need to be careful to not block any other UI on staked balances.

**Comment by marcinbodnar** [comment]
> However I think it is necessary to do a follow-up PR to avoid loading staked balances, as this can be slow query depending on the situation. Generally we need to be careful to not block any other UI on staked balances.

@vgrichina Ok, added issue for it: #1580

**Pull request merged**